From f614a3916e2b1cc01795b28668b8b9c32f9d3c4e Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:49:57 -0400
Subject: [tools] perf/record: Fix display of incorrect mmap pages

Message-id: <1408953210-26343-110-git-send-email-jolsa@redhat.com>
Patchwork-id: 88811
O-Subject: [PATCH RHEL7.1 BZ1133083 109/322] perf record: Fix display of incorrect mmap pages
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit 53653d70a0784a997748cc9e315ddf19d310e812
Author: Adrian Hunter <adrian.hunter@intel.com>
Date: Mon Dec 9 15:18:40 2013 +0200

description
===========
'mmap_pages' is 'unsigned int' not 'int' e.g.

  perf record -m2147483648 uname
  Permission error mapping pages.
  Consider increasing /proc/sys/kernel/perf_event_mlock_kb,
  or try again with a smaller value of -m/--mmap_pages.
  (current value: -2147483648)

Fixed:

  perf record -m2147483648 uname
  Permission error mapping pages.
  Consider increasing /proc/sys/kernel/perf_event_mlock_kb,
  or try again with a smaller value of -m/--mmap_pages.
  (current value: 2147483648)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-record.c b/tools/perf/builtin-record.c
index d93e2ee..c1c1200 100644
--- a/tools/perf/builtin-record.c
+++ b/tools/perf/builtin-record.c
@@ -224,7 +224,7 @@ try_again:
           "Consider increasing "
           "/proc/sys/kernel/perf_event_mlock_kb,\n"
           "or try again with a smaller value of -m/--mmap_pages.\n"
-          "(current value: %d)\n", opts->mmap_pages);
+          "(current value: %u)\n", opts->mmap_pages);
    rc = -errno;
   } else {
    pr_err("failed to mmap with %d (%s)\n", errno, strerror(errno));
-- 
1.7.1